index.js

```diff
@@ -126,7 +126,9 @@ function replacePath (path, keys) {
       name: key || index++,
       delimiter: prefix || '/',
       optional: optional,
-      repeat: repeat
+      repeat: repeat,
+      offset: i + (prefix ? prefix.length : 0),
```

Comment:
@izaakschroeder Hey! Did you want to check out https://github.com/pillarjs/path-to-regexp/pull/50? I pulled everything into a `parser` function that anyone can use and exposed it, then re-implemented the `toRegexp` functionality using it. Takes the extra bits and bops into account too, so there's no more hacks needed.